google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/SingleUseTransaction.java

```diff
@@ -460,33 +448,6 @@ private void checkInterrupted() throws InterruptedException {
     }
   }
 
-  /**
-   * Do a fire-and-forget rollback and close the transaction manager. Fire-and-forget is ok as Cloud
-   * Spanner will abort the transaction automatically after a while if the rollback request should
-   * not actually reach Cloud Spanner.
-   */
-  private void fireAndForgetRollbackAndCloseTxManager(final TransactionManager txManager) {
-    Preconditions.checkNotNull(txManager);
-    Preconditions.checkArgument(
-        txManager.getState()
-            == com.google.cloud.spanner.TransactionManager.TransactionState.STARTED);
-    ExecutorService rollbackExecutor =
-        Executors.newSingleThreadExecutor(new ConnectionImpl.DaemonThreadFactory());
-    rollbackExecutor.submit(
-        new Callable<Void>() {
-          @Override
-          public Void call() throws Exception {
-            try {
-              txManager.rollback();
```

Comment:
[nit] If you could update the PR description to state that now we rely on the backend to abort and rollback the transaction rather than explicitly triggering a rollback, that would be great.

Reply:
I've updated the description to better describe the entire change. The change does however not rely on the backend to rollback/abort the transaction. Instead, it executes a rollback statement synchronously. I.e. an update statement that is executed in autocommit mode and that returns an error, will now wait until the rollback call has finished before returning. Previously, the rollback call would be executed using a separate worker thread.

Reply:
Thanks Knut for the explanation and for updating the description.